src/PKSim.Presentation/Presenters/Simulations/PopulationPKAnalysisPresenter.cs

```diff
@@ -68,23 +69,26 @@ public void CalculatePKAnalyses(IPopulationDataCollector populationDataCollector
          //Calculate first global PK that should always be updated
          _globalPKAnalysisPresenter.CalculatePKAnalysis(new[] { simulation });
 
-         var pkParameters = extractPKParameters(populationAnalysis, simulation);
-         if (!pkParameters.Any())
+         var pkParametersCache = extractPKParameters(populationAnalysis, simulation);
+         if (!pkParametersCache.Any())
             return;
 
-         var captionPrefix = populationAnalysis.AllFieldNamesOn(PivotArea.DataArea);
-
-         var pkAnalyses = _pkAnalysesTask.AggregatePKAnalysis(simulation, pkParameters, populationAnalysis.SelectedStatistics, captionPrefix[0]);
-         _allPKAnalysesIndividualPKValues.AddRange(pkAnalyses);
+         pkParametersCache.KeyValues.Each(pkParameters =>
+         {
+            var pkAnalyses = _pkAnalysesTask.AggregatePKAnalysis(simulation, pkParameters.Value, populationAnalysis.SelectedStatistics, pkParameters.Key);
+            _allPKAnalysesIndividualPKValues.AddRange(pkAnalyses);
+         });
 
          updateView();
       }
 
-
-      private IReadOnlyList<QuantityPKParameter> extractPKParameters(PopulationStatisticalAnalysis populationAnalysis, PopulationSimulation populationSimulation)
```

Comment:
This used to make a flat list with all the pkParameters combined. Now this makes a cache with a list per field.